### PR TITLE
sci-mathematics/fann: add fann-9999 ebuild

### PR DESCRIPTION
**Pull request by zasdfgbnm**
add fann-9999

### CODE REVIEW

File: sci-mathematics/fann/fann-9999.ebuild [hunk 1]
```diff
@@ -0,0 +1,32 @@
+EAPI=5
```
Comment (jlec) [inline sci-mathematics/fann/fann-9999.ebuild]: Header missing. Usage of repoman is obligatory.

File: sci-mathematics/fann/fann-9999.ebuild [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+HOMEPAGE="http://leenissen.dk/fann/"
+EGIT_REPO_URI="git://github.com/libfann/fann.git"
+
+LICENSE="LGPL-2.1"
+SLOT="0"
+KEYWORDS="~amd64 ~ppc ~x86"
+IUSE="examples"
+
+RDEPEND=""
+DEPEND="${RDEPEND}"
```
Comment (jlec) [inline sci-mathematics/fann/fann-9999.ebuild]: drop empty.

File: sci-mathematics/fann/fann-9999.ebuild [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+inherit cmake-utils git-2
+
+DESCRIPTION="Fast Artificial Neural Network Library"
+HOMEPAGE="http://leenissen.dk/fann/"
+EGIT_REPO_URI="git://github.com/libfann/fann.git"
+
+LICENSE="LGPL-2.1"
+SLOT="0"
+KEYWORDS="~amd64 ~ppc ~x86"
```
Comment (jlec) [inline sci-mathematics/fann/fann-9999.ebuild]: Drop KEYWORDs from live ebuilds

File: sci-mathematics/fann/fann-9999.ebuild [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+IUSE="examples"
+
+RDEPEND=""
+DEPEND="${RDEPEND}"
+
+S="${WORKDIR}/${MYP}"
+
+src_test() {
+	cd examples
```
Comment (jlec) [inline sci-mathematics/fann/fann-9999.ebuild]: missing die

**Comment by monsieurp** [comment]
In addition to jlec's pointers, we expect the commit message to have the following format: "cat/pkgname: commit message". Thank you in advance for mending your commit.

**Comment by zasdfgbnm** [comment]
Should now all fixed

**Comment by jlec** [comment]
Would it be possible to squash the commits into one?

File: sci-mathematics/fann/fann-9999.ebuild (hunk 1 above)
Comment (jlec) [inline sci-mathematics/fann/fann-9999.ebuild]: Still you didn't use repoman as the header has the wrong dates.

Comment (zasdfgbnm) [inline sci-mathematics/fann/fann-9999.ebuild]: I'm confused. repoman only complains, it don't fix anything. So I simply copy a header from fann-2.2.0.ebuild...

Comment (jlec) [inline sci-mathematics/fann/fann-9999.ebuild]: it's 2016 and not 2012 anymore.

Comment (zasdfgbnm) [inline sci-mathematics/fann/fann-9999.ebuild]: Is 2001–2016 good, as indicated in the foot of gentoo's homepage?

Comment (jlec) [inline sci-mathematics/fann/fann-9999.ebuild]: no, use this one

```
# Copyright 1999-2016 Gentoo Foundation
# Distributed under the terms of the GNU General Public License v2
# $Id$
```
which you find in ```/usr/portage/header.txt```

Comment (zasdfgbnm) [inline sci-mathematics/fann/fann-9999.ebuild]: Thanks

**Comment by zasdfgbnm** [comment]
It should work now

**Comment by jlec** [comment]
* Failed Patch: flann-1.8.4-options.patch !
 *  ( /home/justin/tree/sci-libs/flann/files/flann-1.8.4-options.patch )
 *
 * Include in your bugreport the contents of:
 *
 *   /var/tmp/portage/sci-libs/flann-9999/temp/flann-1.8.4-options.patch.out

 * ERROR: sci-libs/flann-9999::gentoo failed (prepare phase):
 *   Failed Patch: flann-1.8.4-options.patch!
 *
 * Call stack:
 *     ebuild.sh, line  133:  Called src_prepare
 *   environment, line 3569:  Called cmake-utils_src_prepare
 *   environment, line  990:  Called _execute_optionally 'src_prepare'
 *   environment, line  337:  Called enable_cmake-utils_src_prepare
 *   environment, line 1403:  Called epatch '/home/justin/tree/sci-libs/flann/files/flann-1.8.4-options.patch' '/home/justin/tree/sci-libs/flann/files/flann-1.8.4-CUDA_NVCC_FLAGS.patch' '/home/justin/tree/sci-libs/flann/files/flann-1.8.4-cuda5.5.patch' '/home/justin/tree/sci-libs/flann/files/flann-1.8.4-multilib.patch' '/home/justin/tree/sci-libs/flann/files/flann-1.8.4-docdir.patch'
 *   environment, line 1460:  Called epatch '/home/justin/tree/sci-libs/flann/files/flann-1.8.4-options.patch'
 *   environment, line 1638:  Called die
 * The specific snippet of code:
 *               die "Failed Patch: ${patchname}!";
 *

**Comment by zasdfgbnm** [comment]
That's weird. I never apply any patches on the ebuild and it works well on my computer... Why these patches?

**Comment by jlec** [comment]
> That's weird. I never apply any patches on the ebuild and it works well on my computer... Why these patches?

you are right. my ebuild is different. let me check again.

**Comment by jlec** [comment]
alright, I looked at sci-libs/flann.

**Comment by jlec** [comment]
src_test() needs some more tweaking:
```
>>> Test phase: sci-mathematics/fann-9999
make -j12 -l9 'CFLAGS=-march=native -mtune=native -O2 -pipe -frecord-gcc-switches -fdiagnostics-color=always -Wformat -Werror=format-security -I../src/include -L/var/tmp/portage/sci-mathematics/fann-9999/work/fann-9999_build/src'
gcc -I ../src/include -L ../src/ -O3 xor_train.c -o xor_train -lfann -lm
gcc -I ../src/include -L ../src/ -O3 xor_test.c -o xor_test -lfann -lm
gcc -I ../src/include -L ../src/ -O3 -DFIXEDFANN xor_test.c -o xor_test_fixed -lfixedfann -lm
gcc -I ../src/include -L ../src/ -O3 simple_train.c -o simple_train -lfann -lm
gcc -I ../src/include -L ../src/ -O3 steepness_train.c -o steepness_train -lfann -lm
gcc -I ../src/include -L ../src/ -O3 simple_test.c -o simple_test -lfann -lm
gcc -I ../src/include -L ../src/ -O3 robot.c -o robot -lfann -lm
gcc -I ../src/include -L ../src/ -O3 mushroom.c -o mushroom -lfann -lm
gcc -I ../src/include -L ../src/ -O3 cascade_train.c -o cascade_train -lfann -lm
gcc -I ../src/include -L ../src/ -O3 scaling_test.c -o scaling_test -lfann -lm
gcc -I ../src/include -L ../src/ -O3 scaling_train.c -o scaling_train -lfann -lm
/usr/lib/gcc/x86_64-pc-linux-gnu/5.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: cannot find -lfann
collect2: error: ld returned 1 exit status
Makefile:11: recipe for target 'simple_train' failed
make: *** [simple_train] Error 1
make: *** Waiting for unfinished jobs....
/usr/lib/gcc/x86_64-pc-linux-gnu/5.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: cannot find -lfann
collect2: error: ld returned 1 exit status
Makefile:11: recipe for target 'xor_train' failed
make: *** [xor_train] Error 1
/usr/lib/gcc/x86_64-pc-linux-gnu/5.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: cannot find -lfixedfann
collect2: error: ld returned 1 exit status
Makefile:14: recipe for target 'xor_test_fixed' failed
make: *** [xor_test_fixed] Error 1
/usr/lib/gcc/x86_64-pc-linux-gnu/5.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: cannot find -lfann
collect2: error: ld returned 1 exit status
Makefile:11: recipe for target 'scaling_train' failed
make: *** [scaling_train] Error 1
/usr/lib/gcc/x86_64-pc-linux-gnu/5.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: cannot find -lfann
collect2: error: ld returned 1 exit status
Makefile:11: recipe for target 'mushroom' failed
make: *** [mushroom] Error 1
/usr/lib/gcc/x86_64-pc-linux-gnu/5.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: cannot find -lfann
collect2: error: ld returned 1 exit status
Makefile:11: recipe for target 'scaling_test' failed
make: *** [scaling_test] Error 1
/usr/lib/gcc/x86_64-pc-linux-gnu/5.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: cannot find -lfann
collect2: error: ld returned 1 exit status
Makefile:11: recipe for target 'robot' failed
make: *** [robot] Error 1
/usr/lib/gcc/x86_64-pc-linux-gnu/5.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: cannot find -lfann
collect2: error: ld returned 1 exit status
Makefile:11: recipe for target 'xor_test' failed
make: *** [xor_test] Error 1
/usr/lib/gcc/x86_64-pc-linux-gnu/5.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: cannot find -lfann
collect2: error: ld returned 1 exit status
Makefile:11: recipe for target 'simple_test' failed
make: *** [simple_test] Error 1
/usr/lib/gcc/x86_64-pc-linux-gnu/5.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: cannot find -lfann
collect2: error: ld returned 1 exit status
Makefile:11: recipe for target 'cascade_train' failed
make: *** [cascade_train] Error 1
/usr/lib/gcc/x86_64-pc-linux-gnu/5.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: cannot find -lfann
collect2: error: ld returned 1 exit status
Makefile:11: recipe for target 'steepness_train' failed
make: *** [steepness_train] Error 1
```

**Comment by zasdfgbnm** [comment]
That's weird... It worked for me also...

>>> Source compiled.
>>> Test phase: sci-mathematics/fann-9999
make -j5 'CFLAGS=-O2 -pipe -march=native -I../src/include -L/var/tmp/portage/sci-mathematics/fann-9999/work/fann-9999_build/src' 
gcc -I ../src/include -L ../src/ -O3 xor_train.c -o xor_train -lfann -lm
gcc -I ../src/include -L ../src/ -O3 xor_test.c -o xor_test -lfann -lm
gcc -I ../src/include -L ../src/ -O3 -DFIXEDFANN xor_test.c -o xor_test_fixed -lfixedfann -lm
gcc -I ../src/include -L ../src/ -O3 simple_train.c -o simple_train -lfann -lm
gcc -I ../src/include -L ../src/ -O3 steepness_train.c -o steepness_train -lfann -lm
gcc -I ../src/include -L ../src/ -O3 simple_test.c -o simple_test -lfann -lm
gcc -I ../src/include -L ../src/ -O3 robot.c -o robot -lfann -lm
gcc -I ../src/include -L ../src/ -O3 mushroom.c -o mushroom -lfann -lm
gcc -I ../src/include -L ../src/ -O3 cascade_train.c -o cascade_train -lfann -lm
gcc -I ../src/include -L ../src/ -O3 scaling_test.c -o scaling_test -lfann -lm
gcc -I ../src/include -L ../src/ -O3 scaling_train.c -o scaling_train -lfann -lm
make -j5 runtest 

Training network
./xor_train
Creating network.
Training network.
Max epochs     1000. Desired error: 0.0000000000.
Epochs            1. Current error: 0.3667557836. Bit fail 4.
Epochs           10. Current error: 0.0732617155. Bit fail 4.
Epochs           20. Current error: 0.0001642921. Bit fail 2.
Epochs           24. Current error: 0.0000177596. Bit fail 0.
Testing network. 0.000011
XOR test (-1.000000,-1.000000) -> -0.999749, should be -1.000000, difference=0.000251
XOR test (-1.000000,1.000000) -> 0.992781, should be 1.000000, difference=0.007219
XOR test (1.000000,-1.000000) -> 0.989908, should be 1.000000, difference=0.010092
XOR test (1.000000,1.000000) -> -0.995351, should be -1.000000, difference=0.004649
Saving network.
Cleaning up.

Testing network with floats
./xor_test
Creating network.
Layer / Neuron 0123456
L   1 / N    3 Ccb....
L   1 / N    4 CCD....
L   1 / N    5 dCc....
L   1 / N    6 .......
L   2 / N    7 ...DCDB
L   2 / N    8 .......
Input layer                          :   2 neurons, 1 bias
  Hidden layer                       :   3 neurons, 1 bias
Output layer                         :   1 neurons
Total neurons and biases             :   8
Total connections                    :  13
Connection rate                      :   1.000
Network type                         :   FANN_NETTYPE_LAYER
Training algorithm                   :   FANN_TRAIN_RPROP
Training error function              :   FANN_ERRORFUNC_TANH
Training stop function               :   FANN_STOPFUNC_BIT
Bit fail limit                       :   0.010
Learning rate                        :   0.700
Learning momentum                    :   0.000
Quickprop decay                      :  -0.000100
Quickprop mu                         :   1.750
RPROP increase factor                :   1.200
RPROP decrease factor                :   0.500
RPROP delta min                      :   0.000
RPROP delta max                      :  50.000
Cascade output change fraction       :   0.010000
Cascade candidate change fraction    :   0.010000
Cascade output stagnation epochs     :  12
Cascade candidate stagnation epochs  :  12
Cascade max output epochs            : 150
Cascade min output epochs            :  50
Cascade max candidate epochs         : 150
Cascade min candidate epochs         :  50
Cascade weight multiplier            :   0.400
Cascade candidate limit              :1000.000
Cascade activation functions[0]      :   FANN_SIGMOID
Cascade activation functions[1]      :   FANN_SIGMOID_SYMMETRIC
Cascade activation functions[2]      :   FANN_GAUSSIAN
Cascade activation functions[3]      :   FANN_GAUSSIAN_SYMMETRIC
Cascade activation functions[4]      :   FANN_ELLIOT
Cascade activation functions[5]      :   FANN_ELLIOT_SYMMETRIC
Cascade activation functions[6]      :   FANN_SIN_SYMMETRIC
Cascade activation functions[7]      :   FANN_COS_SYMMETRIC
Cascade activation functions[8]      :   FANN_SIN
Cascade activation functions[9]      :   FANN_COS
Cascade activation steepnesses[0]    :   0.250
Cascade activation steepnesses[1]    :   0.500
Cascade activation steepnesses[2]    :   0.750
Cascade activation steepnesses[3]    :   1.000
Cascade candidate groups             :   2
Cascade no. of candidates            :  80
Testing network.
XOR test (-1.000000, -1.000000) -> -0.999749, should be -1.000000, difference=0.000251
XOR test (-1.000000, 1.000000) -> 0.992781, should be 1.000000, difference=0.007219
XOR test (1.000000, -1.000000) -> 0.989908, should be 1.000000, difference=0.010092
XOR test (1.000000, 1.000000) -> -0.995351, should be -1.000000, difference=0.004649
Cleaning up.

Testing network with fixed points
./xor_test_fixed
Creating network.
Layer / Neuron 0123456
L   1 / N    3 Ccb....
L   1 / N    4 CCD....
L   1 / N    5 dCc....
L   1 / N    6 .......
L   2 / N    7 ...DCDB
L   2 / N    8 .......
Input layer                          :   2 neurons, 1 bias
  Hidden layer                       :   3 neurons, 1 bias
Output layer                         :   1 neurons
Total neurons and biases             :   8
Total connections                    :  13
Connection rate                      :   1.000
Network type                         :   FANN_NETTYPE_LAYER
Decimal point                        :  13
Multiplier                           :8192
Bit fail limit                       :  82
Testing network.
XOR test (-8192, -8192) -> -8192, should be -8192, difference=0.000000
XOR test (-8192, 8192) -> 8192, should be 8192, difference=0.000000
XOR test (8192, -8192) -> 8192, should be 8192, difference=0.000000
XOR test (8192, 8192) -> -8192, should be -8192, difference=0.000000
Cleaning up.
make -j5 clean 
rm -f xor_train xor_test xor_test_fixed simple_train steepness_train simple_test robot mushroom cascade_train scaling_test scaling_train xor_train_debug xor_test_debug xor_test_fixed_debug cascade_train_debug xor_fixed.data *.net *~ *.obj *.exe *.tds noscale.txt withscale.txt scale_test_results.txt
>>> Completed testing sci-mathematics/fann-9999

**Comment by zasdfgbnm** [comment]
$ emerge --info
Portage 2.2.26 (python 2.7.10-final-0, default/linux/amd64/13.0/desktop/plasma/systemd, gcc-5.3.0, glibc-2.21-r1, 4.1.6-ck x86_64)
=================================================================
System uname: Linux-4.1.6-ck-x86_64-Intel-R-_Core-TM-_i5-4278U_CPU_@_2.60GHz-with-gentoo-2.2
KiB Mem:     8072880 total,   3246668 free
KiB Swap:    7812092 total,   7812092 free
Timestamp of repository gentoo: Fri, 08 Jan 2016 17:30:01 +0000
sh bash 4.3_p42-r1
ld GNU ld (Gentoo 2.25.1 p1.1) 2.25.1
app-shells/bash:          4.3_p42-r1::gentoo
dev-java/java-config:     2.2.0::gentoo
dev-lang/perl:            5.20.2::gentoo
dev-lang/python:          2.7.10-r1::gentoo, 3.4.3-r1::gentoo
dev-util/cmake:           3.3.1-r1::gentoo
dev-util/pkgconfig:       0.28-r2::gentoo
sys-apps/baselayout:      2.2::gentoo
sys-apps/openrc:          0.18.4::gentoo
sys-apps/sandbox:         2.6-r1::gentoo
sys-devel/autoconf:       2.13::gentoo, 2.69::gentoo
sys-devel/automake:       1.14.1::gentoo, 1.15::gentoo
sys-devel/binutils:       2.25.1-r1::gentoo
sys-devel/gcc:            5.3.0::gentoo
sys-devel/gcc-config:     1.7.3::gentoo
sys-devel/libtool:        2.4.6::gentoo
sys-devel/make:           4.1-r1::gentoo
sys-kernel/linux-headers: 3.18::gentoo (virtual/os-headers)
sys-libs/glibc:           2.21-r1::gentoo
Repositories:

gentoo
    location: /usr/portage
    sync-type: rsync
    sync-uri: rsync://rsync.gentoo.org/gentoo-portage
    priority: -1000

gentoo-zh
    location: /var/lib/layman/gentoo-zh
    sync-type: laymansync
    sync-uri: git://github.com/microcai/gentoo-zh.git
    masters: gentoo
    priority: 50

jorgicio
    location: /var/lib/layman/jorgicio
    sync-type: laymansync
    sync-uri: https://github.com/jorgicio/jorgicio-gentoo.git
    masters: gentoo
    priority: 50

kde
    location: /var/lib/layman/kde
    sync-type: laymansync
    sync-uri: git://anongit.gentoo.org/proj/kde.git
    masters: gentoo
    priority: 50

lif
    location: /var/lib/layman/lif
    sync-type: laymansync
    sync-uri: https://github.com/killer2tester/gentoo-overlay-lif.git
    masters: gentoo
    priority: 50

qt
    location: /var/lib/layman/qt
    sync-type: laymansync
    sync-uri: git://anongit.gentoo.org/proj/qt.git
    masters: gentoo
    priority: 50

science
    location: /var/lib/layman/science
    sync-type: laymansync
    sync-uri: git://anongit.gentoo.org/proj/sci.git
    masters: gentoo
    priority: 50

zasdfgbnm-overlay
    location: /var/lib/layman/zasdfgbnm-overlay
    sync-type: laymansync
    sync-uri: https://github.com/zasdfgbnm/zasdfgbnm-overlay.git
    masters: gentoo
    priority: 50

brother-overlay
    location: /usr/local/portage/brother-overlay
    sync-type: git
    sync-uri: https://github.com/stefan-langenmaier/brother-overlay.git
    masters: gentoo
    priority: 9999

ACCEPT_KEYWORDS="amd64"
ACCEPT_LICENSE="*"
CBUILD="x86_64-pc-linux-gnu"
CFLAGS="-O2 -pipe -march=native"
CHOST="x86_64-pc-linux-gnu"
CONFIG_PROTECT="/etc /usr/lib64/libreoffice/program/sofficerc /usr/share/config /usr/share/gnupg/qualified.txt /var/spool/torque"
CONFIG_PROTECT_MASK="/etc/ca-certificates.conf /etc/dconf /etc/env.d /etc/fonts/fonts.conf /etc/gconf /etc/gentoo-release /etc/php/apache2-php5.6/ext-active/ /etc/php/cgi-php5.6/ext-active/ /etc/php/cli-php5.6/ext-active/ /etc/revdep-rebuild /etc/sandbox.d /etc/terminfo /etc/texmf/language.dat.d /etc/texmf/language.def.d /etc/texmf/updmap.d /etc/texmf/web2c"
CXXFLAGS="-O2 -pipe -march=native"
DISTDIR="/usr/portage/distfiles"
FCFLAGS="-O2 -pipe"
FEATURES="assume-digests binpkg-logs config-protect-if-modified distlocks ebuild-locks fixlafiles merge-sync news parallel-fetch preserve-libs protect-owned sandbox sfperms strict unknown-features-warn unmerge-logs unmerge-orphans userfetch userpriv usersandbox usersync xattr"
FFLAGS="-O2 -pipe"
GENTOO_MIRRORS="http://distfiles.gentoo.org"
LANG="en_US.utf8"
LDFLAGS="-Wl,-O1 -Wl,--as-needed"
MAKEOPTS="-j5"
PKGDIR="/usr/portage/packages"
PORTAGE_CONFIGROOT="/"
PORTAGE_RSYNC_OPTS="--recursive --links --safe-links --perms --times --omit-dir-times --compress --force --whole-file --delete --stats --human-readable --timeout=180 --exclude=/distfiles --exclude=/local --exclude=/packages"
PORTAGE_TMPDIR="/var/tmp"
USE="X a52 aac acl acpi alsa amd64 apache2 berkdb bluetooth branding bzip2 cairo cdda cdr cli cracklib crypt cups cxx dbus declarative dri dts dvd dvdr emboss encode exif fam firefox flac fortran gdbm gif glamor gpm gtk iconv icu ipv6 ipython jpeg kde kipi lapack latex lcms ldap libnotify mad mmx mmxext mng modules mp3 mp4 mpeg multilib ncurses networkmanager nls nptl ogg opengl openmp pam pango pcre pdf phonon php plasma png policykit ppds pulseaudio qml qt3support qt4 qt5 readline samba sdl seccomp semantic-desktop session spell sqlite sse sse2 ssl startup-notification svg systemd tcl tcpd tiff tk truetype udev udisks unicode upower usb vim-syntax vorbis wayland widgets wxwidgets x264 xattr xcb xcomposite xetex xinerama xml xscreensaver xv xvid zlib zsh-completion" ABI_X86="64" ALSA_CARDS="ali5451 als4000 atiixp atiixp-modem bt87x ca0106 cmipci emu10k1x ens1370 ens1371 es1938 es1968 fm801 hda-intel intel8x0 intel8x0m maestro3 trident usb-audio via82xx via82xx-modem ymfpci" APACHE2_MODULES="authn_core authz_core socache_shmcb unixd actions alias auth_basic authn_alias authn_anon authn_dbm authn_default authn_file authz_dbm authz_default authz_groupfile authz_host authz_owner authz_user autoindex cache cgi cgid dav dav_fs dav_lock deflate dir disk_cache env expires ext_filter file_cache filter headers include info log_config logio mem_cache mime mime_magic negotiation rewrite setenvif speling status unique_id userdir usertrack vhost_alias" CALLIGRA_FEATURES="kexi words flow plan sheets stage tables krita karbon braindump author" CAMERAS="ptp2" COLLECTD_PLUGINS="df interface irq load memory rrdtool swap syslog" CPU_FLAGS_X86="aes avx avx2 fma3 mmx mmxext popcnt sse sse2 sse3 sse4_1 sse4_2 ssse3" ELIBC="glibc" GPSD_PROTOCOLS="ashtech aivdm earthmate evermore fv18 garmin garmintxt gpsclock itrax mtk3301 nmea ntrip navcom oceanserver oldstyle oncore rtcm104v2 rtcm104v3 sirf superstar2 timing tsip tripmate tnt ublox ubx" INPUT_DEVICES="evdev mtrack synaptics" KERNEL="linux" LCD_DEVICES="bayrad cfontz cfontz633 glk hd44780 lb216 lcdm001 mtxorb ncurses text" LIBREOFFICE_EXTENSIONS="presenter-console presenter-minimizer" OFFICE_IMPLEMENTATION="libreoffice" PHP_TARGETS="php5-5" PYTHON_SINGLE_TARGET="python2_7" PYTHON_TARGETS="python2_7 python3_4" RUBY_TARGETS="ruby20 ruby21" USERLAND="GNU" VIDEO_CARDS="intel i965 vesa virtualbox" XTABLES_ADDONS="quota2 psd pknock lscan length2 ipv4options ipset ipp2p iface geoip fuzzy condition tee tarpit sysrq steal rawnat logmark ipmark dhcpmac delude chaos account"
Unset:  CC, CPPFLAGS, CTARGET, CXX, EMERGE_DEFAULT_OPTS, INSTALL_MASK, LC_ALL, PORTAGE_BUNZIP2_COMMAND, PORTAGE_COMPRESS, PORTAGE_COMPRESS_FLAGS, PORTAGE_RSYNC_EXTRA_OPTS, USE_PYTHON

**Comment by jlec** [comment]
There are several issues in the test, which need to be fixed in the build system:
* It shouldn't call gcc, but respect CC
* It shouldn't use -O3 but CFLAGS
* It should respect LDFLAGS
* You probably have already installed flann so the linking works. But you need to make the build system link from the newly build libs

**Comment by zasdfgbnm** [comment]
It should work now. 
In the makefile, gcc and -O3 is explicitly specified. So we should assume that the upstream only plan to support gcc officially, and these test code should be written in a way that works well under -O3.

File: sci-mathematics/fann/fann-9999.ebuild [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+HOMEPAGE="http://leenissen.dk/fann/"
+EGIT_REPO_URI="git://github.com/libfann/fann.git"
+
+LICENSE="LGPL-2.1"
+SLOT="0"
+IUSE="examples"
+
+src_test() {
+	cd examples || die 'fails to enter examples directory'
+    LD_LIBRARY_PATH="${BUILD_DIR}/src" GCC="${CHOST}-gcc ${CFLAGS} -I../src/include -L${BUILD_DIR}/src" emake -e runtest
```
Comment (jlec) [inline sci-mathematics/fann/fann-9999.ebuild]: replace ```${CHOST}-gcc``` by ```tc-getCC``` from ```toolchain-funcs.eclass```

**Comment by zasdfgbnm** [comment]
fixed

**Pull request merged**